projects/storefrontlib/src/cms-components/navigation/search-box/search-box.component.ts

```diff
@@ -89,6 +113,15 @@ export class SearchBoxComponent {
     switchMap((config) => this.searchBoxComponentService.getResults(config))
   );
 
+  ngOnInit() {
+    this.subscription = this.routingService
+      .getRouterState()
+      .subscribe((data) => {
+        if (!data.nextState && data.state.params?.query !== this.chosenWord)
```

Comment:
1) Any reason to keep the `!data.nextState`? What's the use case for it?

2) As mentioned before, wouldn't it be better to include `data.state.context.type === PageType.CONTENT_PAGE && data.state.context.id === 'search'` in order to be sure that you're really in the search page. 
As I'm not sure if xxx.params?.query is the only use case for the search page. This property may exist in other places. (not sure, but it's a possibility)